commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AdvancedFilter.java

```diff
@@ -0,0 +1,103 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+/**
+ * Helper class which holds information advanced-filter username to dn in ldap authorization of security realm.
+ */
+public final class AdvancedFilter {
+
+    private final String baseDn;
+    private final String filter;
+    private final Boolean force;
+    private final Boolean recursive;
+    private final String userDnAttribute;
+    private final LdapCache cache;
+
+    private AdvancedFilter(Builder builder) {
+        this.baseDn = builder.baseDn;
+        this.filter = builder.filter;
+        this.force = builder.force;
+        this.recursive = builder.recursive;
+        this.userDnAttribute = builder.userDnAttribute;
+        this.cache = builder.cache;
+    }
+
+    public String getBaseDn() {
+        return baseDn;
+    }
+
+    public String getFilter() {
+        return filter;
+    }
+
+    public Boolean getForce() {
+        return force;
+    }
+
+    public Boolean getRecursive() {
+        return recursive;
+    }
+
+    public String getUserDnAttribute() {
+        return userDnAttribute;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```

Comment:
Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.